pkg/cloudprovider/vsphereparavirtual/apis/nsxnetworking/v1alpha1/staticroute_types.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// StaticRoute is the Schema for the staticroutes API.
+type StaticRoute struct {
```

Comment:
Do these new CRDs of NSX VPC get stores in a central place? Or Cloud-provider-vsphere have to define these new APIs here and get consumed by NCP or other component? This seems weird

Reply:
Here: https://github.com/vmware-tanzu/nsx-operator/tree/main/pkg/apis
nsx-operator defines them (but legacy v1 ipppol and routeset CRD are still in private nsx-ujo repo). I think we can import them. But the clients/listers/informers are still in dev branch https://github.com/vmware-tanzu/nsx-operator/tree/vpc_dev/pkg/client. It seems they won't merge the dev branch in short time before the project gets stable.
So I copied the CRD definations like previous CPI did for v1 ippool and routeset and generate ourselves so far..